core_contracts/loans/loans/positions.py

```diff
@@ -369,7 +463,8 @@ def _take_snapshot(self) -> None:
                 snapshot.prices[symbol] = assets[symbol].priceInLoop()
         snapshot.snap_time.set(self._loans.now())
         self._loans.Snapshot(self._loans.getDay())
-        self._snapshot_db.start_new_snapshot()
+        if self._loans.getDay() != self._loans._continuous_reward_day.get():
```

Comment:
@dbrehmer I think we should have < rather than != here. as we want to take snapshot before we switch to continuous rewards day.